diff --git a/op-node/rollup/derive/system_config.go b/op-node/rollup/derive/system_config.go
index 72c4e713c..f63157e29 100644
--- a/op-node/rollup/derive/system_config.go
+++ b/op-node/rollup/derive/system_config.go
@@ -142,6 +142,9 @@ func ProcessSystemConfigUpdateLogEvent(destSysCfg *eth.SystemConfig, ev *types.L
 		destSysCfg.GasLimit = gasLimit
 		return nil
 	case SystemConfigUpdateEIP1559Params:
+	        if !rollupCfg.IsHolocene(l1Time) {
+		        return nil // ignore event prior to Holocene
+		}
 		if pointer, err := solabi.ReadUint64(reader); err != nil || pointer != 32 {
 			return NewCriticalError(errors.New("invalid pointer field"))
 		}
