Subject: [PATCH] Lisk Sepolia patch
---
Index: op-node/rollup/derive/system_config.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/op-node/rollup/derive/system_config.go b/op-node/rollup/derive/system_config.go
--- a/op-node/rollup/derive/system_config.go	(revision bd0ef87f20bc2575c060571b92de9b4ce249fac5)
+++ b/op-node/rollup/derive/system_config.go	(revision 8c0a1fc612c062a61253cb2763796f1bffe95e89)
@@ -142,6 +142,9 @@
 		destSysCfg.GasLimit = gasLimit
 		return nil
 	case SystemConfigUpdateEIP1559Params:
+		if !rollupCfg.IsHolocene(l1Time) {
+			return nil // ignore event prior to Holocene
+		}
 		if pointer, err := solabi.ReadUint64(reader); err != nil || pointer != 32 {
 			return NewCriticalError(errors.New("invalid pointer field"))
 		}
